### PR TITLE
Docu for java tracing

### PR DESCRIPTION
**Pull request by rherrmannr**
fix #2438

### CODE REVIEW

File: plugins/org.yakindu.sct.doc.user/src/user-guide/code_generation_java.textile [hunk 1]
```diff
@@ -827,6 +827,65 @@ public static void main(String[] args) {
...
+	 */
+	void onStateEntered(T state);
+	
+	/**
+	* OnStateExited will be called if any state has been exited.
+	*/
+	void onStateExited(T state);
+}
+
+p. With additional generated API in the generated state machine, listeners can be added to and removed from the state machine. Therefore, a implementation of the _ITracingListener_ interface is required. A minimum implementation could be done like this:
```
Comment (BeckmaR) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/code_generation_java.textile]: _an_ implementation :)

File: plugins/org.yakindu.sct.doc.user/src/user-guide/code_generation_java.textile [hunk 2]
```diff
@@ -827,6 +827,65 @@ public static void main(String[] args) {
...
+
+bc.. 
+public class TracingImpl<T> implements ITracingListener<T>{
+	@Override
+	public void onStateEntered(T state) {}
+	@Override
+	public void onStateExited(T state) {}
+}
+
+p. If the tracing feature is enabled, at least one listener must be added:
```
Comment (BeckmaR) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/code_generation_java.textile]: Why? Is that checked somewhere? There's nothing wrong with iterating over an empty list.

File: plugins/org.yakindu.sct.doc.user/src/user-guide/code_generation_java.textile [hunk 3]
```diff
@@ -827,6 +827,65 @@ public static void main(String[] args) {
...
+public class StatemachineRunner {
+	public void runStatemachine() {
+		SomeStatemachine sm = new SomeStatemachine();
+		
+		// create multiple observer
+		TracingImpl<State> ifaceTraceObserver1 = new TracingImpl<State>();
+		TracingImpl<State> ifaceTraceObserver2 = new TracingImpl<State>();
+		
+		// add observer to the statemachine
+		sm.addIfaceTraceObserver(ifaceTraceObserver1);
```
Comment (andreasmuelder) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/code_generation_java.textile]: we should call the methods addTraceObserver and removeTraceObserver

File: plugins/org.yakindu.sct.doc.user/src/user-guide/code_generation_java.textile [hunk 4]
```diff
@@ -827,6 +827,65 @@ public static void main(String[] args) {
...
+
+p. If the tracing feature is enabled, one or more listeners can be added:
+
+bc.. 
+public class StatemachineRunner {
+	public void runStatemachine() {
+		SomeStatemachine sm = new SomeStatemachine();
+		
+		// create multiple observer
+		TracingImpl<State> ifaceTraceObserver1 = new TracingImpl<State>();
```
Comment (andreasmuelder) [inline plugins/org.yakindu.sct.doc.user/src/user-guide/code_generation_java.textile]: rename it to traceObserver1 and traceObserver2

**Pull request merged**